File: backend-v2/src/main/java/vn/io/vutiendat3601/shop/v2/order/OrderPaymentDto.java
package vn.io.vutiendat3601.shop.v2.order;

import java.math.BigDecimal;
import java.time.ZonedDateTime;
import vn.io.vutiendat3601.shop.v2.payment.PaymentStatus;

public record OrderPaymentDto(
    String trackingNumber,
    String ref,
    BigDecimal amount,
    PaymentStatus status,
    String message,
    String errorMessage,
    String paymentUrl,
    ZonedDateTime paymentUrlExpiredAt,
    String callbackUrl) {}

File: frontend-v2/src/app/domain/order/order-payment-dto.ts
export class OrderPaymentDto {
  constructor(
    public trackingNumber: string,
    public ref: string,
    public amount: number,
    public status: string,
    public message: string,
    public errorMessage: string,
    public paymentUrl: string,
    public paymentUrlExpiredAt: string,
    public callbackUrl: string
  ) {}
}
